Proteomics/RetentionTimePrediction/SSRCalc3.cs

```diff
@@ -0,0 +1,1486 @@
+                cc = cc.ReplaceAAs("AMYV", "1"); // Not L10N
+                cc = cc.ReplaceAAs("A-Z", "0"); // Not L10N
+            }
+            else
+            // Suppress the unreachable code warning
+#pragma warning disable 162
+            // ReSharper disable HeuristicUnreachableCode
+            {
+                cc = cc.ReplaceAAs("LIWF", "5"); // Not L10N
+                cc = cc.ReplaceAAs("MYV", "1"); // Not L10N
+                cc = cc.ReplaceAAs("A-Z", "0"); // Not L10N
+            }
+            // ReSharper restore HeuristicUnreachableCode
+#pragma warning restore 162
+
+            double score = 0.0;
+            //
+            // Translator1 note:  check on true meaning of the algorithm that defines 'occurs'
+            // Should an encoded aa string such as 015101510 match pick "01510" once or twice?
+            // The perl code seems to match once.  0151001510 would match twice.
+
+            foreach (var pair in CLUSTCOMB)
+            {
+                int occurs = 0;
+                Match m = pair.Key.Match(cc);
+                while (m.Success)
+                {
+                    occurs++;
+                    m = m.NextMatch();
+                }
+                if (occurs > 0)
+                {
+                    double sk = pair.Value;
+                    double addit = sk * occurs;
+                    score += addit;
+                }
+            }
+            return score * KSCALE;
+        }
+
+        // ============================================================
+        //  process based on proline - v 2,3 algorithm
+        private static double proline(String sq)
+        {
+            double score = 0.0;
+            if (sq.Contains("PPPP")) // Not L10N
+                score = PPPPSCORE;
+            else if (sq.Contains("PPP")) // Not L10N
+                score = PPPSCORE;
+            else if (sq.Contains("PP")) // Not L10N
+                score = PPSCORE;
+            return score;
+        }
+
+        // ============================================================
+        //  scaling based on length - v 1,2,3 algorithms
+        private static double length_scale(int sqlen)
+        {
+            double LS = 1.0;
+            if (sqlen < SPLim)
+            {
+                LS = 1.0 + SPSFac * (SPLim - sqlen);
+            }
+            else
+            {
+                if (sqlen > LPLim)
+                {
+                    LS = 1.0 / (1.0 + LPSFac * (sqlen - LPLim));
+                }
+            }
+            return LS;
+        }
+
+        private static int eMap(char aa)
+        {
+            switch (aa)
+            {
+                case 'K': return 0;
+                case 'R': return 1;
+                case 'H': return 2;
+                case 'D': return 3;
+                case 'E': return 4;
+                case 'C': return 5;
+                case 'Y': return 6;
+                default: return -1;
```

Comment:
if-else is faster (according to Stefan), in case this is looped through a lot

Reply:
I've made this an array instead of a case or ifelse, so it should be a lookup now.